src/javaharness/java/arcs/crdt/CollectionDataTest.java

```diff
@@ -0,0 +1,292 @@
+        CollectionOpTypes.ADD,
+        new Data("two"),
+        new VersionMap(){{ put("me", 2); }},
+        "me"
+    ));
+
+    verifyIds(set, new String[]{"one", "two"});
+  }
+
+  public static void testSameValueTwoActors() {
+    // can add the same value from two actors
+    CRDTCollection<Data> set = new CRDTCollection<>();
+    assert set.applyOperation(new CollectionOperation<Data>(
+        CollectionOpTypes.ADD,
+        new Data("one"),
+        new VersionMap() {{ put("me", 1); }},
+        "me"
+    ));
+    assert set.applyOperation(new CollectionOperation<Data>(
+        CollectionOpTypes.ADD,
+        new Data("one"),
+        new VersionMap() {{ put("them", 1); }},
+        "them"
+    ));
+
+    verifyIds(set, new String[]{"one"});
+  }
+
+  public static void testRejectAddsNotInSequence() {
+    // rejects add operations not in sequence
+    CRDTCollection<Data> set = new CRDTCollection<>();
+    assert set.applyOperation(new CollectionOperation<Data>(
+      CollectionOpTypes.ADD,
+      new Data("one"),
+      new VersionMap() {{ put("me", 1); }},
+      "me"
+    ));
+    assert !set.applyOperation(new CollectionOperation<Data>(
+      CollectionOpTypes.ADD,
+      new Data("two"),
+      new VersionMap() {{ put("me", 0); }},
+      "me"
+    ));
+    assert !set.applyOperation(new CollectionOperation<Data>(
+      CollectionOpTypes.ADD,
+      new Data("two"),
+      new VersionMap() {{ put("me", 1); }},
+      "me"
+    ));
+    assert !set.applyOperation(new CollectionOperation<Data>(
+      CollectionOpTypes.ADD,
+      new Data("two"),
+      new VersionMap() {{ put("me", 3); }},
+      "me"
+    ));
+  }
+
+  public static void testRemoveItem() {
+    // can remove an item
+    CRDTCollection<Data> set = new CRDTCollection<>();
+    assert set.applyOperation(new CollectionOperation<Data>(
+      CollectionOpTypes.ADD,
+      new Data("one"),
+      new VersionMap() {{ put("me", 1); }},
+      "me"
+    ));
+    verifySize(set, 1);
+    assert set.applyOperation(new CollectionOperation<Data>(
+      CollectionOpTypes.REMOVE,
+      new Data("one"),
+      new VersionMap() {{ put("me", 1); }},
+      "me"
+    ));
+    verifySize(set, 0);
+  }
+
+  public static void testRejectRemoveIfVersionMismatch() {
+    // rejects remove operations if version mismatch
+    CRDTCollection<Data> set = new CRDTCollection<>();
+    assert set.applyOperation(new CollectionOperation<Data>(
+      CollectionOpTypes.ADD,
+      new Data("one"),
+      new VersionMap() {{ put("me", 1); }},
+      "me"
+    ));
+    assert !set.applyOperation(new CollectionOperation<Data>(
+      CollectionOpTypes.REMOVE,
+      new Data("one"),
+      new VersionMap() {{ put("me", 2); }},
+      "me"
+    ));
+    assert !set.applyOperation(new CollectionOperation<Data>(
+      CollectionOpTypes.REMOVE,
+      new Data("one"),
+      new VersionMap() {{ put("me", 0); }},
+      "me"
+    ));
+  }
+
+  public static void testRejectRemoveNonexitent() {
```

Comment:
s/exitent/existent/ here and below where referenced

Reply:
done